### DISCUSSION TITLE
Singular and Plural terms

### DISCUSSION THREAD

**Discussion by seanrockster** [ravendb/ravendb#18205] [General]
Hi

We have Polo (shirts) products. Some customers search for Polo and others search for Polos. The term Polos exists in only a few of the description and marketing info so the results are different.

Is there a way to automatically generate singular and plural forms of a term or would I have to explicitly add those?

Thanks.

**Comment by DejanMilicic** [comment]
Are you talking about implementing full-text search, where for every occurrence of "polo", you would also populate "polos" inside of an index?

**Reply by seanrockster** [reply]
Yes, so for a given term, e.g. Polo, the index would contain Polo and Polos, or given Polos, the index would contain Polos and Polo.

**Comment by ayende** [comment]
Here is how you can do this:

https://ayende.com/blog/200833-B/using-analyzers-in-ravendb-indexing?key=4956877853d24b69ae578dbca2afbc09

**Reply by seanrockster** [reply]
@ayende   you're awesome! Thanks!

**Reply by seanrockster** [reply]
@ayende which version of raven is this possible with? The version in use is 3 and the app is a framework 4.8 and not sure the customer is interested in upgrading the .core. Thanks.

**Reply by ppekrol** [reply]
@ayende was referring to version 4 and up, in version 3 we had a notion of plugins https://ravendb.net/docs/article-page/3.5/csharp/server/plugins/analyzer-generators

**Reply by seanrockster** [reply]
@ppekrol thats great, thanks!

**Reply by DejanMilicic** [reply]
@seanrockster here is working sample
https://github.com/DejanMilicic/RavenWorkshop/tree/main/Northwind/Northwind/Features/Analyzers/Snowball
